### ISSUE TITLE
Chips

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Starship-EVO#2775]
I'm rather worried about the new addition of chips in the game. Besides being super powerful which can always be balanced, it reminds me a bit too much of Avorion that had a similar system and not for the better.
https://avorion.gamepedia.com/System_Upgrade
These upgrades would enhance your ship eventually to such a degree that the ship would be invincible. The game no longer becomes who can build the best ship, but who has the best arbitrary upgrade set. Even worse, SEVO chips go on your person rather than in your ship. Here is a some suggestions to improve the chip system.

1. Keep the same random loot system, but have it specific to a single system. You can add a chip to a weapon or a system rather than impacting the entire ship. This removes some of the problem, but still has the issue that effectiveness is based on chips rather than construction. At least now the AI can actually use them. All ideas below use the same per system upgrades.

2. Make it sort of like Rust where rather than getting chips, you get blueprints to craft them beyond the basic stuff. So if you want to have a more effective chip, you need to buy or loot to research to craft upgraded chips. This adds elements of progression and adds to a future SCIENCE mechanic, but it still has some of the problems above.

3.  Instead of just getting chips, you craft/buy an "empty" chip (wafer?) This has no upgrades on it, but rather you add these on yourself. Using blueprints like 2, you can learn different upgrades you can then add onto the chip yourself. If the tier system must remain, it can go two ways. First is the wafers themselves can have upgrades where higher tier wafers can have more boosts on it. Second would be keeping all wafers the same, but the upgrades learned have better stats depending on rarity. This fixes most of the problems and would be difficult enough to acquire better stats or wafers that it would sort of balance it out.

3.5 While back you talked about your horizontal upgrade philosophy for weapons where all weapons will have a purpose regardless of how advanced the ship or technology is. The current chip system is very much vertical. Why would you use a basic tier chip when an epic one is available and better? And no, simple energy requirement won't be enough to balance it unless the upgrades are low. Using 3 as a base, I suggest a Stellaris like trait system to balance out positives and negative. In game when you are picking your faction's traits, you can only use so many positive without negatives to balance it out. https://stellaris.paradoxwikis.com/Traits . In accompanying it, Poor can have one positive traits, Common can have two positive and one drawback, Uncommon can have three positives and two drawbacks, and so on. This keeps things horizontal and fair for all and solves the problems above.

**Comment by tsunamayo** [comment]
I havent had time to play Avorion, but yes I guess that's the route I want to go, more focused on campaign progression rather than hardcore survival a la Space Engineer (which I havent played too).
Chips go on to your ship, also I will spawn some chip on each NPCs.
Are you talking in a PVP type?
Without gameplay elements like that (crew and hull tier level will be another), then it will be who has the bigger ship anyway. By having this I am essentially making smaller ship more powerful versus bigger (and lower tier) one.
I think I will have to drop the horizontal upgrade scheme for weapons, it is not interesting, and I need something for gameplay progression.
Also I dont see how your suggestions fixes anything, you are just moving the issue elsewhere.
They are valid I though about it before, but opt to not to (1 is too tedious, 2 is overkill at this stage ...)
Cheers

**Comment by ultrak2k** [comment]
> I havent had time to play Avorion, but yes I guess that's the route I want to go, more focused on campaign progression rather than hardcore survival a la Space Engineer (which I havent played too).
> Chips go on to your ship, also I will spawn some chip on each NPCs.
> Are you talking in a PVP type?
> Without gameplay elements like that (crew and hull tier level will be another), then it will be who has the bigger ship anyway. By having this I am essentially making smaller ship more powerful versus bigger (and lower tier) one.
> I think I will have to drop the horizontal upgrade scheme for weapons, it is not interesting, and I need something for gameplay progression.
> Also I dont see how your suggestions fixes anything, you are just moving the issue elsewhere.
> They are valid I though about it before, but opt to not to (1 is too tedious, 2 is overkill at this stage ...)
> Cheers

With the horizontal upgrade scheme, is this types being inferior? like beam is worse then laser? Or kinetics being worse? I hope it's neither, personally, and it's upgrades of each weapon type

also with hull tiers, is this flat out upgrades? or light/heavy armour variants?
will we have different hull types, say some that resist heat better but are weak in general? and if so, will those tier up too?

**Comment by tsunamayo** [comment]
No I want to make each weapon type a valid option at any point in the game progression, I am talking more about the side upgrade, to have them tied to the game progression.
For Hull I wanted to use a ladder with light hull and armored one (but heavy), then you do an alloy of both metals to get the proper Higher tier one.

**Comment by ultrak2k** [comment]
> No I want to make each weapon type a valid option at any point in the game progression, I am talking more about the side upgrade, to have them tied to the game progression.
> For Hull I wanted to use a ladder with light hull and armored one (but heavy), then you do an alloy of both metals to get the proper Higher tier one.

wait, will there be genuine metal mixing to discover your own hull alloys?

**Comment by tsunamayo** [comment]
No I cant afford all combination possible for alloy as I am limited  in term of how many Hull type I can store in the voxel data. Plus it is a bit pointless.

**Comment by ultrak2k** [comment]
ah okay, fair

**Comment by tsunamayo** [comment]
@ZachZent btw you also seemed to imply chip are a bad aspect of Avorion, is it because of the invincibility issue only?
Can you stack stats in Avorion? Ie you get two shield boost do you get say +100% + 50% = +150% (or maybe even + 200% if they are trully stacked). In SEVO you cant stack so it dont become too op (knowing that crew will also give some boost)
Cheers

**Comment by Kaiser-Indrasil** [comment]
Yes, there is chip effect stacking in Abortion, however there's no shield boosting chip as far as I can remember.

**Comment by ZachZent** [comment]
Sorry for the delay Tsuna and everyone else. I just moved apartments and so had to take apart all my computer parts. Indrasil answered that question, but seeing the relation to Avorion brought on a few alarm bells. Avorion isn't a bad game, but there are a number of design choices I heavily disagree with. Alone the chips aren't so bad, but combined with a lot of other decisions like a VERY rigid vertical tier system for almost everything in game. Such as building materials which against is compounded by a "race for the center" gameplay where you need to go to the center for better materials.
